buildkite/resource_pipeline_test.go

```diff
@@ -658,4 +666,86 @@ func TestAccBuildkitePipelineResource(t *testing.T) {
 			},
 		})
 	})
+
+	t.Run("create a pipeline with signed steps", func(t *testing.T) {
+		const (
+			repository = "my-repo"
+			jwksKeyID  = "my-key"
+		)
+
+		pipelineName := acctest.RandString(12)
+
+		privateJWKS, _, err := jwkutil.NewKeyPair(jwksKeyID, jwa.EdDSA)
+		if err != nil {
+			t.Fatalf("Failed to generate key pair: %v", err)
+		}
+
+		privateKey, ok := privateJWKS.Key(0)
+		if !ok {
+			t.Fatalf("Failed to get private key from JWKS")
+		}
+
+		jwks, err := json.Marshal(privateKey)
+		if err != nil {
+			t.Fatalf("Failed to marshal private key: %v", err)
+		}
+
+		p, err := pipeline.Parse(strings.NewReader(defaultSteps))
+		if err != nil {
+			t.Fatalf("Failed to parse pipeline: %v", err)
+		}
+
+		if err != signature.SignPipeline(p, privateKey, repository) {
+			t.Fatalf("Failed to sign pipeline: %v", err)
+		}
+
+		signedSteps, err := yaml.Marshal(p)
+		if err != nil {
+			t.Fatalf("Failed to marshal signed steps: %v", err)
+		}
+		resource.ParallelTest(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: protoV6ProviderFactories(),
+			Steps: []resource.TestStep{
+				{
+					Config: heredoc.Docf(
+						`
+							data "buildkite_signed_pipeline_steps" "my_signed_steps" {
+							  repository     = %q
+							  jwks           = %q
+							  jwks_key_id    = %q
+							  unsigned_steps = %q
+							}
+							resource "buildkite_pipeline" "pipeline" {
+								name = "%s"
+								repository = "https://github.com/buildkite/terraform-provider-buildkite.git"
+								steps = data.buildkite_signed_pipeline_steps.my_signed_steps.steps
+							}
+						`,
+						repository,
+						jwks,
+						jwksKeyID,
+						defaultSteps,
+						pipelineName,
+					),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttrWith("buildkite_pipeline.pipeline", "steps", func(value string) error {
+							var err error
+							// if the steps attribute value still matches the default steps, it has not been signed so
+							// there is a bug
+							if value == defaultSteps {
+								err = fmt.Errorf("Pipeline steps have not been signed: %q...%q", value, signedSteps)
+							}
+							return err
+						}),
+						resource.TestCheckResourceAttr(
+							"buildkite_pipeline.pipeline",
+							"steps",
+							string(signedSteps),
+						),
```

Comment:
I need some extra eyes here. I'm trying to create a test for this case but can't reproduce it. This step passes already.

I expected it to fail because it should be using the default unsigned steps that get applied to a pipeline if no steps are specified. But I have verified locally that the generated steps for the pipeline are the signed steps from the datasource.